agent/agent.go

```diff
@@ -29,6 +31,21 @@ func NewAgent(ef *registry.ExtractorFactory, pf *registry.ProcessorFactory, sf *
 	}
 }
 
+func (r *Agent) Validate(rcp recipe.Recipe) error {
```

Comment:
Does it make sense to make it part of recipe? Because that is what we are validating.

Reply:
we can't because recipe package does not depend on plugins and only agent currently has all the plugin factories.